mock/mock.go

```diff
@@ -694,38 +694,49 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 
 	for i := 0; i < maxArgCount; i++ {
 		var actual, expected interface{}
-		var actualFmt, expectedFmt string
 
 		if len(objects) <= i {
 			actual = "(Missing)"
-			actualFmt = "(Missing)"
 		} else {
 			actual = objects[i]
-			actualFmt = fmt.Sprintf("(%[1]T=%[1]v)", actual)
 		}
 
 		if len(args) <= i {
 			expected = "(Missing)"
-			expectedFmt = "(Missing)"
 		} else {
 			expected = args[i]
-			expectedFmt = fmt.Sprintf("(%[1]T=%[1]v)", expected)
+		}
+
+		// %v formatting of the actual or expected object must be deferred to avoid race conditions
+		// Use these methods to defer %v until needed
+		actualFmt := func(actual interface{}) string {
+			if len(objects) <= i {
+				return "(Missing)"
+			}
+			return fmt.Sprintf("(%[1]T=%[1]v)", actual)
+		}
+
+		expectedFmt := func(expected interface{}) string {
+			if len(args) <= i {
+				return "(Missing)"
+			}
+			return fmt.Sprintf("(%[1]T=%[1]v)", expected)
 		}
 
 		if matcher, ok := expected.(argumentMatcher); ok {
 			if matcher.Matches(actual) {
-				output = fmt.Sprintf("%s\t%d: PASS:  %s matched by %s\n", output, i, actualFmt, matcher)
+				output = fmt.Sprintf("%s\t%d: PASS:  %s matched by %s\n", output, i, actualFmt(actual), matcher)
 			} else {
 				differences++
-				output = fmt.Sprintf("%s\t%d: FAIL:  %s not matched by %s\n", output, i, actualFmt, matcher)
+				output = fmt.Sprintf("%s\t%d: PASS:  %s not matched by %s\n", output, i, actualFmt(actual), matcher)
```

Comment:
shouldn't this still be a `FAIL` instead of a `PASS`?

Reply:
Yes